databases/functions/userFollowUser/user-follow-add.js

```diff
@@ -21,22 +21,28 @@ module.exports = async (
   let follows_array = [];
   await Promise.all(
     followedUsers.map(async (val) => {
-      //follow sign user
-      follows_array.push({
-        follow_action_id: uuidv4(),
-        user_id_follower: userId,
-        user_id_followed: val
-      });
-
       const userFollowed = await UsersFunction.findUserById(User, val);
       if (userFollowed.is_anonymous === false) {
+        follows_array.push({
+          follow_action_id: uuidv4(),
+          user_id_follower: userId,
+          user_id_followed: val
+        });
+
         const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
         follows_array.push({
           follow_action_id: uuidv4(),
           user_id_follower: userId,
           user_id_followed: anonymousUser?.user_id,
           is_anonymous: true
         });
+      } else if (userFollowed.is_anonymous === true) {
+        follows_array.push({
+          follow_action_id: uuidv4(),
+          user_id_follower: userId,
+          user_id_followed: val,
+          is_anonymous: true
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for following anonymous users seems to be incorrect. If the user is not anonymous, you are adding a follow action for both the non-anonymous user and an associated anonymous user. However, if the user is anonymous, you only add a follow action for the anonymous user. This could lead to inconsistencies in your data. 

Also, there's no error handling for the case when `UsersFunction.findUserById` or `UsersFunction.findAnonymousUserId` fails. It would be better to handle these potential errors to prevent unexpected behavior.

Here's a suggested fix:

```diff
   let follows_array = [];
   await Promise.all(
     followedUsers.map(async (val) => {
        try {
+          const userFollowed = await UsersFunction.findUserById(User, val);
+          if (!userFollowed) {
+            throw new Error(`User with id ${val} not found`);
+          }
           if (userFollowed.is_anonymous === false) {
             follows_array.push({
               follow_action_id: uuidv4(),
               user_id_follower: userId,
               user_id_followed: val
             });
 
-            const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
-            follows_array.push({
-              follow_action_id: uuidv4(),
-              user_id_follower: userId,
-              user_id_followed: anonymousUser?.user_id,
-              is_anonymous: true
-            });
           } else if (userFollowed.is_anonymous === true) {
             follows_array.push({
               follow_action_id: uuidv4(),
               user_id_follower: userId,
               user_id_followed: val,
               is_anonymous: true
             });
           }
+        } catch (error) {
+          console.error(`Failed to follow user: ${error.message}`);
+        }
     })
   );
```
This code will throw an error if the user is not found and only add a follow action for non-anonymous users. It also adds error handling for potential failures when finding users.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
mas @brm-stnd ini ada kemungkinan user anon sudah di follow duluan sebelum follow sign user, kalau kondisi seperti itu bukanya nanti kalau kita follow sign user jadi duplikat untuk data follow anonnya?

Reply:
oh iya mas, sudah saya perbaiki ya mas